Here are files and codes used in this project and the output files generated

GCOS_Cel contains the orginal input files (the roscovitine treated samples) as well as some output files generated during the limma search

RNASeq.txt is an output file created as a result of the limma analysis which was then read into the DESeq2 analysis

annotResults.txt is an output file created as a result of running the DESeq2 analysis

example.R contains the code used to run the limma analysis in RStudio

phylo_R_studio_report_code.R contains the code used to create a tree from an correctly aligned .fasta file in RStudio. The trees were constructed outside of Linux and the images could not be uploaded to Github.

RNASeq.R contains the code used to run the DESeq2 analysis in RStudio.
